### DISCUSSION TITLE
SwiftUI Animation API Support

### DISCUSSION THREAD

**Discussion by DominatorVbN** [skiptools/skip#9] [Q&A] [answered]
As of know skip does not support animation APIs of SwiftUI to Android, AFAIK.

**Comment by DominatorVbN** [comment]
Too many arguments for public fun withAnimation(body: () -> Unit): Unit defined in skip.ui

Not entirely correct just saw that withAnimation works just the additional params like curve do not.

**Comment by DominatorVbN** [comment]
https://github.com/skiptools/skip-ui/blob/main/Sources/SkipUI/SkipUI/Animation/Animation.swift

Ahh... Pretty confused now, the API exist to Crash... !?

**Reply by DominatorVbN** [reply]
https://github.com/skiptools/skip-ui/blob/main/Sources/SkipUI/SkipUI/Animation/Animation.swift
Answered!

**Comment by marcprux** [comment, marked as answer]
> Ahh... Pretty confused now, the API exist to Crash... !?

I assume you are referencing that `withAnimation` just calls `fatalError()` at:

https://github.com/skiptools/skip-ui/blob/3a7ccb00d368d9b1858f55719ced660e7f0e4ee1/Sources/SkipUI/SkipUI/Animation/Animation.swift#L181-L182

The current state of Skip is that a number of unsupported API calls simply crash. In many cases, we have annotated these with `@unavailable`, so at the least the Skip transpiler plugin will warn you about using unsupported API. However, in this case, we didn't, and hence your crash.

We will work on getting better availability annotations implemented in the near future. Thanks for your patience.
